provisioning/README.md

```diff
@@ -0,0 +1,109 @@
+# Mbed Crypto provisioning example application
+
+Let's say the secure element manufacturer shipped you a secure element with a
+device private key already inside. This type of "pre-provisioned" secure
+element is what you'll use as a part during the device assembly process. You've
+taken this secure element, paired it with a nice Cortex-M microcontroller on a
+board to make a device. Now, you'd like to be able to use this key from Mbed
+Crypto. How does Mbed Crypto know the key exists? We can tell Mbed Crypto by
+running a provisionining application, which is typically separate from the
+primary application and may be run as a final device assembly step.
+
+
+### Adding pre-provisioned keys
```

Comment:
While “Adding” is correct, I think it would be less confusing to consistently use “Registering”.